File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        i =0
        maxi=0
        vowel=0
        vowels = {'a','e','i','o','u'}

        for ids in range(k):
            if(s[ids] in vowels):
                vowel+=1
        maxi = vowel

        for j in range(k,len(s)):
            if(s[i] in vowels):
                vowel-=1
            i+=1
            if(s[j] in vowels):
                vowel+=1
            maxi = max(maxi,vowel)
        return maxi

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
    bool isVowel(char c){
        return c == 'a' || c== 'e' || c== 'i' || c=='o' || c=='u';
    }

public:
    int maxVowels(string s, int k) {
        int i=0;
        int vowel = 0;
        int maxi =0;

        for(int idx = 0; idx < k ; idx++){
            if(isVowel(s[idx])){
                vowel++;
            }
        }
        maxi = vowel;

        for(int j=k ; j<s.size() ; j++){

            if(isVowel(s[i])){
                vowel--;
            }
            i++;
            if(isVowel(s[j])){
                vowel++;
            }
            maxi= max(maxi,vowel);
        }
        return maxi;
    }

};